opentelemetry-exporter-gcp-logging/src/opentelemetry/exporter/cloud_logging/__init__.py

```diff
@@ -106,15 +106,27 @@
 INVALID_LOG_NAME_MESSAGE = "%s is not a valid log name. log name must be <512 characters and only contain characters: A-Za-z0-9/-_."
 
 
-def _convert_any_value_to_string(value: Any) -> str:
+class _GenAiJsonEncoder(json.JSONEncoder):
+    def default(self, o: Any) -> Any:
+        if isinstance(o, bytes):
+            return b64encode(o).decode()
+        return super().default(o)
+
+
+def _convert_any_value_to_string(value: Any, debug_location: str) -> str:
     if isinstance(value, bool):
         return "true" if value else "false"
     if isinstance(value, bytes):
         return base64.b64encode(value).decode()
     if isinstance(value, (int, float, str)):
         return str(value)
-    if isinstance(value, (list, tuple)):
-        return json.dumps(value)
+    if isinstance(value, (list, tuple, Mapping)):
+        return json.dumps(value, separators=(",", ":"), cls=_GenAiJsonEncoder)
```

Comment:
WDYT of falling back to `str()` to avoid dropping things on the floor? Not sure if this would ever happens because IIRC the sdk does validate the `AnyValue` before it gets enqueued